src/registerAllocator/RegisterAllocator.class.ts

```diff
@@ -1500,7 +1511,9 @@ export class RegisterAllocator {
     Object.keys(this._allocations)
       .filter(matchArg)
       .forEach((v) => {
-        delete this._allocations[v];
+        if (isMem(this._allocations[v].store)) {
```

Comment:
Why do we need to check this? Were we deleting register-values here?

Reply:
I believe we were, specifically registers containing a copy of `arg1[3]` or similar.

Reply:
then I really like this change :D